### PR TITLE
Add availability property & default icon 

### PR DESCRIPTION
**Pull request by mattsaxon**
Note, this change requires later version of pysonofflan (not committed at this time).

It also removes the now redundant call to update which is now invoked via a callback from pysonofflan. This means that switching only occurs once the switch has responded.

### CODE REVIEW

File: manifest.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+        "domain": "sonoff_lan_mode",
+        "name": "Sonoff LAN Mode",
```
Comment (beveradb) [inline manifest.json:3]: I tweaked these slightly to fix typo and add dependency

File: switch.py [hunk 2]
```diff
@@ -11,17 +11,19 @@
...
 from homeassistant.components.switch import (SwitchDevice, PLATFORM_SCHEMA)
-from homeassistant.const import CONF_HOST, CONF_NAME
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_ICON
 
 REQUIREMENTS = ['pysonofflan>=0.2.1']
 
 _LOGGER = logging.getLogger('homeassistant.components.switch.sonoff_lan_mode')
 
 DEFAULT_NAME = 'Sonoff Switch'
+DEFAULT_ICON = 'mdi:flash'
```
Comment (beveradb) [inline switch.py:21]: Thanks, much better

**Review by beveradb — APPROVED** [review]

**Comment by beveradb** [comment]
LGTM, happy to merge once the current work on pysonofflan is complete and we've released a new version to pypi etc.

**Comment by mattsaxon** [comment]
Now you've updated the PySonoffLan component, I've updated the requirement line in switch.py and I think this is ready to merge.

**Review by beveradb — APPROVED** [review]
LGTM

**Pull request merged**